lib/at_host/at_host.c

```diff
@@ -70,49 +70,25 @@ static inline void write_uart_string(const char *str)
 	}
 }
 
-static void response_handler(void *context, const char *response)
+static void response_handler(const char *response)
 {
-	ARG_UNUSED(context);
-
 	/* Forward the data over UART */
 	write_uart_string(response);
 }
 
 static void cmd_send(struct k_work *work)
 {
-	char              str[25];
-	enum at_cmd_state state;
 	int               err;
 
 	ARG_UNUSED(work);
 
-	err = at_cmd_write(at_buf, at_buf,
-			   sizeof(at_buf), &state);
+	err = nrf_modem_at_cmd(at_buf, sizeof(at_buf), "%s", at_buf);
```

Comment:
simply: `nrf_modem_at_cmd(at_buf, sizeof(at_buf), at_buf)`

Reply:
More importantly, we need to escape the `%` character when found in the input, perhaps that can be done in `uart_rx_handler`

Reply:
We don't need special treatment for the `%` character with the original version (and I chose this option in modem_shell because of that):
`nrf_modem_at_cmd(at_buf, sizeof(at_buf), "%s", at_buf);`

But if you have this then you need to escape the `%` character:
`nrf_modem_at_cmd(at_buf, sizeof(at_buf), at_buf)`

Up to you guys which way to go.

Reply:
Then we definitely want to keep using the 1st option here, so we can avoid having to escape the `%` characters afterwards.

Reply:
@MirkoCovizzi we should call `err = nrf_modem_at_cmd(at_buf, sizeof(at_buf), "%s", at_buf);` otherwise we'd have to escape `%` chars like @trantanen pointed out. I feel it's easier to handle it like that, and perhaps leave a comment so we'll know why it's not simply sending the buffer contents.